packages/react-router/lib/rsc/server.ssr.tsx

```diff
@@ -111,12 +112,27 @@ export function RSCStaticRouter({
 
   if (payload.type !== "render") return null;
 
+  let patchedLoaderData = { ...payload.loaderData };
+  for (const match of payload.matches) {
+    if (
+      shouldHydrateRouteLoader(
+        match.id,
+        match.clientLoader,
+        match.hasLoader,
+        false
+      ) &&
+      (match.hydrateFallbackElement || !match.hasLoader)
```

Comment:
Added this line to match up with https://github.com/remix-run/react-router/blob/main/packages/react-router/lib/dom/ssr/server.tsx#L66